From 77be79eead084f965198da2ac525aa73e68bf45b Mon Sep 17 00:00:00 2001
From: David Lechner <david@lechnology.com>
Date: Wed, 14 Jan 2015 22:39:22 -0600
Subject: [PATCH] Fix bluetooth plugin implemntation of technology_set returns
 error.

The technology_set implementation in the bluetooth plugin returns
-EINPROGRESS which causes problems.

1.  When setting the Tethering property via D-Bus, D-Bus returns an error
    instead of completing successfuly.
2.  When using the PersistentTetheringMode option, bluetooth tethering
    is not remembered. This is because /var/lib/connman/settings is not
    updated if technology_set returns an error.

This fixes the issue by returning 0 when bluetooth_tech_set_tethering
completes successfully.
---
 plugins/bluetooth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/bluetooth.c b/plugins/bluetooth.c
index c6f387e..636c7bf 100644
--- a/plugins/bluetooth.c
+++ b/plugins/bluetooth.c
@@ -899,7 +899,7 @@ static int bluetooth_tech_set_tethering(struct connman_technology *technology,
 	if (i == 0)
 		return -ENODEV;
 
-       return -EINPROGRESS;
+	return 0;
 }
 
 static struct connman_technology_driver tech_driver = {
-- 
1.9.1

